packages/bottom-tabs/src/views/BottomTabBar.tsx

```diff
@@ -440,7 +440,7 @@ export function BottomTabBar({
               key={route.key}
               value={descriptors[route.key].navigation}
             >
-              <NavigationRouteContext.Provider value={route}>
+              <NavigationRouteContextProvider value={route}>
```

Comment:
I added a `NavigationProvider` so we won't need to change all of these places.

Let's move all context for route and navigation there to simplify things.